platform/windows/export/template_modifier.cpp

```diff
@@ -0,0 +1,584 @@
+	strings.append(StringStructure(p_key, p_value));
+}
+
+TemplateModifier::StringTable::StringTable() {
+	key = "040904b0";
+	type = 1;
+}
+
+TemplateModifier::StringFileInfo::StringFileInfo() {
+	key = "StringFileInfo";
+	value_length = 0;
+	type = 1;
+}
+
+Vector<uint8_t> TemplateModifier::StringFileInfo::save() const {
+	Vector<uint8_t> bytes = Structure::save();
+	bytes.append_array(string_table.save());
+	return add_length(bytes);
+}
+
+Vector<uint8_t> TemplateModifier::Var::save() const {
+	Vector<uint8_t> bytes = Structure::save();
+	ByteStream::save(value, bytes);
+	return add_length(bytes);
+}
+
+TemplateModifier::Var::Var() {
+	value_length = 4;
+	key = "Translation";
+}
+
+Vector<uint8_t> TemplateModifier::VarFileInfo::save() const {
+	Vector<uint8_t> bytes = Structure::save();
+	bytes.append_array(var.save());
+	return add_length(bytes);
+}
+
+TemplateModifier::VarFileInfo::VarFileInfo() {
+	type = 1;
+	key = "VarFileInfo";
+}
+
+Vector<uint8_t> TemplateModifier::VersionInfo::save() const {
+	Vector<uint8_t> fixed_file_info = value.save();
+	Vector<uint8_t> bytes = Structure::save();
+	bytes.append_array(fixed_file_info);
+	bytes.append_array(string_file_info.save());
+	while (bytes.size() % 4) {
+		bytes.append(0);
+	}
+	bytes.append_array(var_file_info.save());
+	return add_length(bytes);
+}
+
+TemplateModifier::VersionInfo::VersionInfo() {
+	key = "VS_VERSION_INFO";
+	value_length = 52;
+}
+
+Vector<uint8_t> TemplateModifier::IconEntry::save() const {
+	Vector<uint8_t> bytes;
+	ByteStream::save(width, bytes);
+	ByteStream::save(height, bytes);
+	ByteStream::save(colors, bytes);
+	ByteStream::save((uint8_t)0, bytes);
+	ByteStream::save(planes, bytes);
+	ByteStream::save(bits_per_pixel, bytes);
+	ByteStream::save(image_size, bytes);
+	ByteStream::save((uint16_t)image_offset, bytes);
+	return bytes;
+}
+
+void TemplateModifier::IconEntry::load(Ref<FileAccess> p_file) {
+	width = p_file->get_8(); // Width in pixels.
+	height = p_file->get_8(); // Height in pixels.
+	colors = p_file->get_8(); // Number of colors in the palette (0 - no palette).
+	p_file->get_8(); // Reserved.
+	planes = p_file->get_16(); // Number of color planes.
+	bits_per_pixel = p_file->get_16(); // Bits per pixel.
+	image_size = p_file->get_32(); // Image data size in bytes.
+	image_offset = p_file->get_32(); // Image data offset.
+}
+
+Vector<uint8_t> TemplateModifier::GroupIcon::save() const {
+	Vector<uint8_t> bytes;
+	ByteStream::save(reserved, bytes);
+	ByteStream::save(type, bytes);
+	ByteStream::save(image_count, bytes);
+	for (IconEntry icon_entry : icon_entries) {
+		bytes.append_array(icon_entry.save());
+	}
+	return bytes;
+}
+
+void TemplateModifier::GroupIcon::load(Ref<FileAccess> p_icon_file) {
+	if (p_icon_file->get_32() != 0x10000) { // Wrong reserved bytes
+		print_error("Wrong icon file type.");
+		return;
+	}
+	if (p_icon_file->get_16() != IMAGE_COUNT) {
+		print_error("Wrong number of icon images.");
```

Comment:
Is it possible like with rcedit to also do that operation with just one icon image? Would be nice for backward compatibility.